tests/test_ec2/test_elastic_block_store.py

```diff
@@ -53,6 +53,45 @@ def test_create_and_delete_volume():
     cm.exception.request_id.should_not.be.none
 
 
+@mock_ec2_deprecated
+def test_delete_attached_volume():
+    conn = boto.ec2.connect_to_region("us-east-1")
+    reservation = conn.run_instances("ami-1234abcd")
+    # create an instance
+    instance = reservation.instances[0]
+    # create a volume
+    volume = conn.create_volume(80, "us-east-1a")
+    # attach volume to instance
+    volume.attach(instance.id, "/dev/sdh")
+
+    volume.update()
+    volume.volume_state().should.equal("in-use")
+    volume.attachment_state().should.equal("attached")
+
+    volume.attach_data.instance_id.should.equal(instance.id)
+
+    # attempt to delete volume
+    # assert raises VolumeInUseError
+    with assert_raises(EC2ResponseError) as ex:
+        volume.delete()
+    ex.exception.error_code.should.equal("VolumeInUse")
+    ex.exception.status.should.equal(400)
+    ex.exception.message.should.equal(
+        f"Volume {volume.id} is currently attached to {instance.id}"
```

Comment:
Hi @mickeypash - we're still supporting Python2, so f-strings are not supported. Please use the string.format method for this.
Looks good to me otherwise.

Reply:
Oh sorry, will fix that now. I checked the [Docker](https://github.com/spulec/moto/blob/master/Dockerfile#L15) for the Python version which threw me off.